libs/sdk-ui-dashboard/src/model/store/storeAccessors/DashboardStoreAccessorRepository.ts

```diff
@@ -0,0 +1,140 @@
+// (C) 2022 GoodData Corporation
+
+import { DashboardDispatch, DashboardSelectorEvaluator, DashboardState } from "../types";
+import { invariant } from "ts-invariant";
+import { DashboardStoreAccessor } from "./DashboardStoreAccessor";
+import { idRef, ObjRef, serializeObjRef } from "@gooddata/sdk-model";
+
+/**
+ * This singleton class serves the selector and the dispatcher properties for given dashboard {@link @gooddata/sdk-model#IdentifierRef}.
+ *
+ * The {@link Dashboard} component has an optional property {@link IDashboardProps#onStateChange} through which
+ * you can handle setting of the values for {@link DashboardDispatch} and {@link DashboardSelectorEvaluator}.
+ *
+ *
+ * In your component using {@link Dashboard} you can create an onStateChange callback for your dashboard using
+ * {@link DashboardStoreAccessorRepository#getOnChangeHandlerForDashboard} and pass it to mentioned Dashboard
+ * component property.
+ *
+ * @example
+ * ```
+ *  const dashboardStoreAccessor = DashboardStoreAccessorRepository.getInstance();
+ *
+ *
+ *  // in the code where needed
+ *  dashboardStoreAccessor.getAccessorsForDashboard(<DASHBOARD_ID>).getDispatch()(
+ *      changeDateFilterSelection("relative", "GDC.time.month", "-3", "0"),
+ *  );
+ *
+ *  // or with check if accessor is initialized already
+ *  if (dashboardStoreAccessor.isAccessorInitializedForDashboard(DASHBOARD_ID)) {
+ *          setSelectResult(
+ *              dashboardStoreAccessor.getAccessorsForDashboard(DASHBOARD_ID).getSelector()(
+ *                  selectEffectiveDateFilterOptions,
+ *              ),
+ *          );
+ *      }
+ *
+ *  return (
+ *      <Dashboard dashboard={"<dashboardId>"} onStateChange={dashboardStoreAccessor.getOnChangeHandlerForDashboard(DASHBOARD_REF)}/>
+ *  )
+ * ```
+ *
+ * To get latest properties, use static member function {@link DashboardStoreAccessor#getInstance}. If there is already an instance
+ * created, it will return this instance and will return new instance of the {@link DashboardStoreAccessor} otherwise.
+ *
+ * @public
+ */
+export class DashboardStoreAccessorRepository {
+    static dashboardAccessor: DashboardStoreAccessorRepository;
```

Comment:
This should probably be private to avoid unintentional manipulation from the outside, right?